File: src/types/aiAssistant.ts
// src/types/aiAssistant.ts

export interface AIMessage {
  id: string;
  role: 'user' | 'assistant' | 'system';
  content: string;
  timestamp: string;
  type?: 'text' | 'data' | 'error' | 'suggestion';
  data?: any;
  emotion?: 'happy' | 'neutral' | 'concerned' | 'excited';
}

export interface AIUserContext {
  userId: string;
  userName: string;
  userRole: string;
  accountType: string;
  lastLogin: string;
  preferences: UserPreferences;
  healthProfile?: HealthProfile;
}

export interface UserPreferences {
  language: string;
  voiceEnabled: boolean;
  theme: 'light' | 'dark';
  fontSize: 'small' | 'medium' | 'large';
}

export interface HealthProfile {
  bloodGroup?: string;
  age?: number;
  gender?: string;
  medicalConditions?: string[];
  allergies?: string[];
}

export interface AIResponse {
  message: string;
  type: 'text' | 'data' | 'suggestion' | 'error';
  data?: any;
  suggestions?: string[];
  emotion?: string;
  sqlQuery?: string;
  confidence: number;
}

export interface AIConversation {
  id: string;
  userId: string;
  title: string;
  messages: AIMessage[];
  createdAt: string;
  updatedAt: string;
}

export interface AISuggestion {
  id: string;
  text: string;
  category: 'health' | 'appointment' | 'medicine' | 'general';
  icon: string;
}

export interface AIVoiceConfig {
  isListening: boolean;
  isSpeaking: boolean;
  transcript: string;
  audioUrl?: string;
}

export interface AIDataDisplay {
  type: 'table' | 'card' | 'list';
  title: string;
  data: any;
  columns?: string[];
}



// src/types/aiAssistant.ts

export interface AIMessage {
  id: string;
  role: 'user' | 'assistant' | 'system';
  content: string;
  timestamp: string;
  type?: 'text' | 'data' | 'error' | 'suggestion';
  data?: any;
  emotion?: 'happy' | 'neutral' | 'concerned' | 'excited';
}

export interface AIUserContext {
  userId: string;
  userName: string;
  userRole: string;
  accountType: string;
  lastLogin: string;
  preferences: UserPreferences;
  healthProfile?: HealthProfile;
}

export interface UserPreferences {
  language: string;
  voiceEnabled: boolean;
  theme: 'light' | 'dark';
  fontSize: 'small' | 'medium' | 'large';
}

export interface HealthProfile {
  bloodGroup?: string;
  age?: number;
  gender?: string;
  medicalConditions?: string[];
  allergies?: string[];
}

export interface AIResponse {
  message: string;
  type: 'text' | 'data' | 'suggestion' | 'error';
  data?: any;
  suggestions?: string[];
  emotion?: string;
  sqlQuery?: string;
  confidence: number;
}

export interface AIConversation {
  id: string;
  userId: string;
  title: string;
  messages: AIMessage[];
  createdAt: string;
  updatedAt: string;
}

export interface AISuggestion {
  id: string;
  text: string;
  category: 'health' | 'appointment' | 'medicine' | 'general';
  icon: string;
}

export interface AIVoiceConfig {
  isListening: boolean;
  isSpeaking: boolean;
  transcript: string;
  audioUrl?: string;
}

export interface AIDataDisplay {
  type: 'table' | 'card' | 'list';
  title: string;
  data: any;
  columns?: string[];
}

File: models/ai_assistant.py
from sqlalchemy import (
    Column, String, Text, Boolean, DECIMAL, JSON,
    Enum as SQLEnum, ForeignKey
)
from sqlalchemy.dialects.mysql import CHAR
from ..core.database import Base
from .base import BaseModel


# ============================================
# AI CONVERSATIONS
# ============================================
class AIConversation(Base, BaseModel):
    __tablename__ = "ai_conversations"

    user_id = Column(CHAR(36), ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    title = Column(String(255))
    status = Column(SQLEnum("active", "archived", "deleted"), default="active")


# ============================================
# AI MESSAGES
# ============================================
class AIMessage(Base, BaseModel):
    __tablename__ = "ai_messages"

    conversation_id = Column(CHAR(36), ForeignKey("ai_conversations.id", ondelete="CASCADE"), nullable=False, index=True)
    role = Column(SQLEnum("user", "assistant", "system"), nullable=False)
    content = Column(Text, nullable=False)
    type = Column(SQLEnum("text", "data", "suggestion", "error"), default="text")
    data = Column(JSON)
    emotion = Column(SQLEnum("happy", "neutral", "concerned", "excited"))
    confidence = Column(DECIMAL(3, 2))
    suggestions = Column(JSON)
    sql_query = Column(Text)


# ============================================
# AI VOICE SESSIONS
# ============================================
class AIVoiceSession(Base, BaseModel):
    __tablename__ = "ai_voice_sessions"

    user_id = Column(CHAR(36), ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    transcript = Column(Text)
    audio_url = Column(String(500))
    is_listening = Column(Boolean, default=False)
    is_speaking = Column(Boolean, default=False)
